test/app.e2e-spec.ts

```diff
@@ -0,0 +1,269 @@
+      },
+      {
+        ...OPTIONS,
+        params: {
+          ...OPTIONS.params,
+          'fly-id': 'op1',
+        },
+      },
+    );
+  });
+
+  it('Mint ERC20 token', async () => {
+    const request: TokenMint = {
+      amount: '2',
+      data: 'test',
+      operator: IDENTITY,
+      poolId: CONTRACT_ADDRESS,
+      to: '1',
+    };
+    const response: AsyncResponse = {
+      id: '1',
+    };
+
+    http.post = jest.fn(() => new FakeObservable(response));
+
+    await server.post('/mint').send(request).expect(202).expect({ id: '1' });
+
+    expect(http.post).toHaveBeenCalledTimes(1);
+    expect(http.post).toHaveBeenCalledWith(
+      `${BASE_URL}/abis/${CONTRACT_ABI}/${CONTRACT_ADDRESS}/mintWithData`,
+      {
+        amount: '2',
+        data: '0x74657374',
+        to: '1',
+      },
+      OPTIONS,
+    );
+  });
+
+  it('Transfer token', async () => {
+    const request: TokenTransfer = {
+      amount: '2',
+      from: '1',
+      operator: IDENTITY,
+      poolId: CONTRACT_ADDRESS,
+      to: '2',
+    };
+    const response: EthConnectAsyncResponse = {
+      id: '1',
+      sent: true,
+    };
+
+    http.post = jest.fn(() => new FakeObservable(response));
+
+    await server.post('/transfer').send(request).expect(202).expect({ id: '1' });
+
+    expect(http.post).toHaveBeenCalledTimes(1);
+    expect(http.post).toHaveBeenCalledWith(
+      `${BASE_URL}/abis/${CONTRACT_ABI}/${CONTRACT_ADDRESS}/transferWithData`,
+      {
+        amount: '2',
+        data: '0x00',
+        from: '1',
+        to: '2',
+      },
+      OPTIONS,
+    );
+  });
+
+  it('Burn token', async () => {
+    const request: TokenBurn = {
+      amount: '2',
+      data: 'tx1',
+      from: 'A',
+      operator: IDENTITY,
+      poolId: CONTRACT_ADDRESS,
+    };
+    const response: AsyncResponse = {
+      id: '1',
+    };
+
+    http.post = jest.fn(() => new FakeObservable(response));
+
+    await server.post('/burn').send(request).expect(202).expect({ id: '1' });
+
+    expect(http.post).toHaveBeenCalledTimes(1);
+    expect(http.post).toHaveBeenCalledWith(
+      `${BASE_URL}/abis/${CONTRACT_ABI}/${CONTRACT_ADDRESS}/burnWithData`,
+      {
+        data: '0x747831',
+        from: 'A',
+        amount: '2',
+      },
+      OPTIONS,
+    );
+  });
+
+  it('Query balance', async () => {
+    const request: TokenBalanceQuery = {
+      account: '1',
+      poolId: CONTRACT_ADDRESS,
+    };
+    const response: EthConnectReturn = {
+      output: '1',
+    };
+
+    http.get = jest.fn(() => new FakeObservable(response));
+
+    await server
+      .get('/balance')
+      .query(request)
+      .expect(200)
+      .expect(<TokenBalance>{
+        balance: '1',
+      });
+
+    expect(http.get).toHaveBeenCalledTimes(1);
+    expect(http.get).toHaveBeenCalledWith(
+      `${BASE_URL}/abis/${CONTRACT_ABI}/${CONTRACT_ADDRESS}/balanceOf`,
+      {
+        params: {
+          account: '1',
+        },
+      },
+    );
+  });
+});
```

Comment:
Would like to see more E2E tests here, particularly around parsing emitted websocket events.